tests/unittests/OnnxImporterTest.cpp

```diff
@@ -1585,6 +1585,31 @@ TEST(onnx, importCastToInt64) {
   importCast("castToInt64.onnxtxt", "data", {1, 2, 2, 2}, ElemKind::Int64ITy);
 }
 
+TEST(onnx, cast_32_64) {
+  ExecutionEngine EE{};
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  std::string netFilename(GLOW_DATA_PATH
+                          "tests/models/onnxModels/castInt-32-64.onnxtxt");
+  PlaceholderBindings bindings;
+  Placeholder *graphOutputVar;
+  {
+    Tensor data(ElemKind::FloatTy, {1, 2, 4, 3});
+    ONNXModelLoader onnxLD(netFilename, {"input"}, {&data.getType()}, *F);
+    graphOutputVar = EXIT_ON_ERR(onnxLD.getSingleOutput());
+    bindings.allocate(mod.getPlaceholders());
+    updateInputPlaceholdersByName(bindings, &mod, {"input"}, {&data});
+  }
+
+  EE.compile(CompilationMode::Infer, F);
+  EE.run(bindings);
+  auto result = bindings.get(graphOutputVar)->getHandle();
+  std::vector<size_t> expectedDims = {1, 2, 4, 3};
+
+  EXPECT_TRUE(result.dims().vec() == expectedDims);
```

Comment:
Might need to make testing better here (checking values, etc). Checking just dims seems insufficient.

Reply:
Actually this was supposed just to check that the model loads, but you're right, I'll add more checks.

Reply:
Surprisingly, this test isn't working. Conversion `float->int32->int64->int32->float` works as NOOP when ONNX loads it. Likely it's an underlying bug exposed. Investigating.

Reply:
The bug is in `OptimizeConversions`. It thinks it's a NOOP. Trying to fix it...

Reply:
Will go along with fix.